flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/FileType.java

```diff
@@ -22,6 +22,18 @@
  * Different file types to request from the {@link TaskExecutor}.
  */
 public enum FileType {
+	/**
+	 * the log file type for taskmanager
+	 */
 	LOG,
-	STDOUT
+
+	/**
+	 * the stdout file type for taskmanager
+	 */
+	STDOUT,
+
+	/**
+	 * the thread dump type for taskmanager
+	 */
+	THREAD_DUMP
```

Comment:
IMO, I think it's best to keep it as it is.

Reply:
I disagree since we are not requesting a file here. We are only using the file transfer mechanism to send the data.

The file transfer aspect is another thing we should discuss. Wouldn't it be possible and simpler to directly send the stringified stack trace instead of using the blob service?